### PR TITLE
Logically delete channels

### PR DESCRIPTION
**Pull request by gabichelsea**
Trello card link:
[https://trello.com/c/GxILp1ey](https://trello.com/c/GxILp1ey)

### CODE REVIEW

File: persistence/src/main/java/com/epam/rft/atsy/persistence/entities/ChannelEntity.java [hunk 1]
```diff
@@ -22,18 +22,14 @@
 @ToString(callSuper = true)
 @Entity
 @Table(name = "channels", schema = "atsy", uniqueConstraints = @UniqueConstraint(columnNames = "name"))
-public class ChannelEntity extends SuperEntity implements java.io.Serializable {
+public class ChannelEntity extends LogicallyDeletableEntity implements java.io.Serializable {
 
-
-  @Column(name = "name", nullable = false, length = 255, unique = true)
```
Comment (lamfalusy) [inline persistence/src/main/java/com/epam/rft/atsy/persistence/entities/ChannelEntity.java:28]: Why did you delete length = 255? Is it part of the task?

Comment (gabichelsea) [reply persistence/src/main/java/com/epam/rft/atsy/persistence/entities/ChannelEntity.java:28]: No, but 255 is the default value, so I think it was not necessary to define again.

File: persistence/src/main/java/com/epam/rft/atsy/persistence/entities/ChannelEntity.java [hunk 2]
```diff
@@ -22,6 +24,7 @@
 @ToString(callSuper = true)
 @Entity
 @Table(name = "channels", schema = "atsy", uniqueConstraints = @UniqueConstraint(columnNames = "name"))
+@NamedQueries(@NamedQuery(name = "ChannelEntity.findAll", query = "SELECT channel FROM ChannelEntity channel WHERE channel.deleted is false"))
```
Comment (lamfalusy) [inline persistence/src/main/java/com/epam/rft/atsy/persistence/entities/ChannelEntity.java]: Use more communicative name. This query is more than a simple findAll.

File: persistence/src/main/java/com/epam/rft/atsy/persistence/repositories/ChannelRepository.java [hunk 3]
```diff
@@ -2,11 +2,17 @@
...
 
 /**
  * Repository that allows operations with channels in database.
  */
 public interface ChannelRepository extends JpaRepository<ChannelEntity, Long> {
 
   ChannelEntity findByName(String channelName);
+
+  @Query
+  List<ChannelEntity> findAll();
```
Comment (lamfalusy) [inline persistence/src/main/java/com/epam/rft/atsy/persistence/repositories/ChannelRepository.java]: See my comment above.

File: web/src/main/webapp/resources/js/atsy-settings.js [hunk 4]
```diff
@@ -64,6 +64,7 @@ $(function () {
     new SettingsForm().init('#channels_section','settings.channels.error.empty');
 });
 
+
```
Comment (lamfalusy) [inline web/src/main/webapp/resources/js/atsy-settings.js]: Delete this empty line.

File: persistence/src/main/java/com/epam/rft/atsy/persistence/entities/ChannelEntity.java (hunk 1 above)
Comment (lamfalusy) [reply persistence/src/main/java/com/epam/rft/atsy/persistence/entities/ChannelEntity.java:28]: All right

File: persistence/src/main/resources/db/migration/schema/V1_18__AlterCandidatesTable.sql [hunk 5]
```diff
@@ -1 +1 @@
-ALTER TABLE candidates DROP INDEX cv_file_name ;
+ALTER TABLE candidates DROP INDEX cv_file_name;
```
Comment (lamfalusy) [inline persistence/src/main/resources/db/migration/schema/V1_18__AlterCandidatesTable.sql]: Revert this modification.
